frontend/src/components/student/AllActivities/CombatTask/CombatTask.js

```diff
@@ -81,69 +78,131 @@ export default function CombatTask() {
     []
   )
 
-  return (
-    <Content>
-      <InfoContainer fluid className='p-0'>
-        {!task && errorMessage === '' ? (
-          <Loader />
-        ) : errorMessage !== '' ? (
-          <p>{errorMessage}</p>
-        ) : (
-          <ActivityCol className='invisible-scroll'>
-            <HeaderCol>
-              <HeaderRow>
-                <ActivityImg src={getActivityImg('TASK')}></ActivityImg>
-                <ActivityType>{getActivityTypeName('TASK')}</ActivityType>
-                <ActivityName>{task.name}</ActivityName>
-              </HeaderRow>
-              <FullDivider />
-            </HeaderCol>
-            <div>
-              <h5>{task.description}</h5>
-              {task.points !== null && (
+  //add overflows where needed
+
+  const Header = () => (
+    <>
+      <ActivityImg src={getActivityImg('TASK')}></ActivityImg>
+      <ActivityType>{getActivityTypeName('TASK')}</ActivityType>
+      <ActivityName>{task.name}</ActivityName>
+    </>
+  )
```

Comment:
Zauważ, że używasz identycznego `Header`a co w `ActivityAssessmentDetails`. Ja bym wszystkie elementy tj `Header`, `VertivalSpacer`, `UserDetails` i te pozostałe pomniejsze "komponenty" przeniósł do nowego pliku, a ten plik umieścił w folderze general. I te elementy zapisał bym jak normalne komponenty z propsem i nie jako arrow function. 
Co do samego `Header`a to dałbym mu propsa i przekazywał activityName odpowiednio